elytron/src/main/java/org/wildfly/extension/elytron/CertificateAuthorityAccountDefinition.java

```diff
@@ -0,0 +1,437 @@
+                case "LETSENCRYPT":
+                    return LETS_ENCRYPT;
+                default:
+                    throw new IllegalArgumentException(name);
+            }
+        }
+    }
+
+    static final StringListAttributeDefinition CONTACT_URLS = new StringListAttributeDefinition.Builder(ElytronDescriptionConstants.CONTACT_URLS)
+            .setRequired(false)
+            .setAllowExpression(true)
+            .setMinSize(1)
+            .setRestartAllServices()
+            .build();
+
+    static final SimpleAttributeDefinition CERTIFICATE_AUTHORITY = new SimpleAttributeDefinitionBuilder(ElytronDescriptionConstants.CERTIFICATE_AUTHORITY, ModelType.STRING, true)
+            .setDefaultValue(new ModelNode(CertificateAuthority.LETS_ENCRYPT.getName()))
+            .setAllowedValues(CertificateAuthority.LETS_ENCRYPT.getName())
+            .setAllowExpression(true)
+            .setRestartAllServices()
+            .build();
+
+    static final SimpleAttributeDefinition KEY_STORE = new SimpleAttributeDefinitionBuilder(ElytronDescriptionConstants.KEY_STORE, ModelType.STRING, false)
+            .setAttributeGroup(ElytronDescriptionConstants.ACCOUNT_KEY)
+            .setMinSize(1)
+            .setRestartAllServices()
+            .setCapabilityReference(KEY_STORE_CAPABILITY, CERTIFICATE_AUTHORITY_ACCOUNT_CAPABILITY, true)
+            .build();
+
+    static final SimpleAttributeDefinition ALIAS = new SimpleAttributeDefinitionBuilder(ElytronDescriptionConstants.ALIAS, ModelType.STRING, false)
+            .setAttributeGroup(ElytronDescriptionConstants.ACCOUNT_KEY)
+            .setMinSize(1)
+            .setAllowExpression(true)
+            .setRestartAllServices()
+            .build();
+
+    static final ObjectTypeAttributeDefinition CREDENTIAL_REFERENCE = CredentialReference.getAttributeBuilder(true, true)
+            .setAttributeGroup(ElytronDescriptionConstants.ACCOUNT_KEY)
+            .build();
+
+    private static final AttributeDefinition[] ATTRIBUTES = new AttributeDefinition[] { CERTIFICATE_AUTHORITY, CONTACT_URLS, KEY_STORE, ALIAS, CREDENTIAL_REFERENCE };
+
+    static final SimpleAttributeDefinition AGREE_TO_TERMS_OF_SERVICE = new SimpleAttributeDefinitionBuilder(ElytronDescriptionConstants.AGREE_TO_TERMS_OF_SERVICE, ModelType.BOOLEAN, false)
+            .setAllowExpression(true)
+            .build();
+
+    static final SimpleAttributeDefinition STAGING = new SimpleAttributeDefinitionBuilder(ElytronDescriptionConstants.STAGING, ModelType.BOOLEAN, true)
+            .setAllowExpression(true)
+            .setDefaultValue(new ModelNode(false))
+            .build();
+
+    private static final AcmeClientSpi acmeClient;
+    static {
+        acmeClient = loadAcmeClient();
+    }
+
+    private static AcmeClientSpi loadAcmeClient() {
+        for (AcmeClientSpi acmeClient : ServiceLoader.load(AcmeClientSpi.class, ElytronSubsystemMessages.class.getClassLoader())) {
```

Comment:
Same question than for AcmeClient.

Can there be more than 1 AcmeClientSpi and does it matter?

Reply:
Same as above, i.e., there will only be one (WildFlyAcmeClient) but if there was more than one, it wouldn't matter because only one is needed to prove ownership of a domain name.